### PR TITLE
Arfecta update / Pug Farming

### PR DESCRIPTION
**Pull request by tehhowch**
<details><summary>old stuff</summary>The Arfecta is woefully outmatched by the player if boarded, and despite being warned about driving off the Pug, there aren't currently any consequences to doing so.

This PR addresses the capturability and combat effectiveness of the Arfecta (significantly increasing the former, and somewhat increasing the latter). A warning is added to foreshadow the permanent rep loss in the form of hostility when the Arfecta are attacked. If the player continues attacking the Arfectas, and ultimately kills or captures both of them, then they will not be able to resume working with the Wanderers.
With the right starting reputation and timings, the player would still be able to obtain 2 Arfectas and complete the Wanderer story (excepting the possibility that the final arc needs the mission Arfectas to be alive).</details>

This PR provides a small increase to the combat effectiveness of the Arfecta (ensuring it can maintain range and avoid its own splash damage, even as hostile ships encroach upon it), and makes fighting the Arfecta more challenging for the player, as the Wanderer ships will engage the player if they attack any in-system Pug ships.

So long as the player does not capture both Ap'arak Arfectas, or attack too many Wanderer ships in the process, there is no permanent damage to player-Wanderer reputation. Pug ships in any other systems remain free game for attack & plunder.

### CODE REVIEW

**Comment by solardawning** [comment]
As a note, this would not prevent the player from capturing the Arfecta that spawns as a hidden, rep-conditional mission.

Currently, this Arfecta can be triggered as long as the player is not on the main (Free Worlds) storyline- either BEFORE joining the Free Worlds, or after completing the campaign.

If attacked and captured BEFORE joining the Free Worlds, the player can take advantage of the Pug reputation reset that occurs at the end of the Free Worlds campaign to get back into good standing with them, and thus have no lasting consequences.

If the goal of this PR is to ensure that capturing an Arfecta = lasting reputation consequences, then I suggest the following:

In Pug.txt, change the mission data for "Pug Angry At Being Farmed" to require has "main plot completed" instead of not "chosen sides".

**Comment by tehhowch** [comment]
>With the right starting reputation and timings, the player would still be able to obtain 2 Arfectas and complete the Wanderer story (excepting the possibility that the final arc needs the mission Arfectas to be alive).

I'm aware that this does not prevent the player from obtaining Arfectas. This PR makes it more challenging to do so, and adds a penalty to your rep with the Wanderers for *doing what they ask you not to do* (in "Tell Wanderers about the Pug").

**Comment by solardawning** [comment]
I was interpreting that the intent was to make sure that there were lasting consequences of some kind for doing so (when also accounting for the recent change to Pug reputation so that players can no longer earn rep with them for killing Unfettered or pirates), so I would have been remiss not to mention that there is still an edge case in which a player can take an Arfecta and still return to positive Pug reputation afterward.
If that's outside of the scope of what you want to do with this PR, I'll create a separate 'Issue' for it.

**Comment by jafdy** [comment]
It's a shame that you can't bake a faster speed into the hull without giving the hull cloak. (As in a cloaking device is still needed but on that one ship the cloak activates faster.)
I think that would be a bit much for one use case (then again we have phasing).

**Comment by geojak** [comment]
The 2 arfectas are spawned by a mission. how about changing it, such that if the player decided to kill, capture an arfectas then the players they will instantly turn hostile to the wanderes

this can be achieved by splitting the current into two mission which both spawn 1 arfecta each plis
on complete
     wanderer reputation = -1

also wanderer ships should not accepts bribes so one cant bribe and farm reputation back up

**Comment by Amazinite** [comment]
>Also, I dont think it makes sense to punish the players reputation with the wanderer for farming pug in pug lyig as the wanderer currently have no way to know

It's already the case under this PR that your rep with the Wanderers wouldn't be affected. The added attitude of the Wanderers toward the Pug will only cause the Wanderers to attack you if you attack the Pug in the same system as them, but your reputation toward the Wanderers will not change.

**Comment by tehhowch** [comment]
>It's already the case under this PR that your rep with the Wanderers wouldn't be affected. The added attitude of the Wanderers toward the Pug will only cause the Wanderers to attack you if you attack the Pug in the same system as them, but your reputation toward the Wanderers will not change.

At .05 rep changes are permanent but slow, given the low crew of Pug ships. So yes, engaging the Pug in any system after they begin protecting Ap'arak will eventually cause the Wanderers to become hostile.

**Comment by temtemy** [comment]
I noticed that after the Arfectas in Ap'arak are destroyed/captured, the Pug didn't cared to replace the two. That doesn't make sense. There should be four new Arfectas replacing the two (four because the player is now a threat, in fact a larger threat than the Unfettered).

**Comment by KiLEdEnNis** [comment]
which just allows even more Arfecta farming, which is gonna be a big no

**Comment by Zitchas** [comment]
I agree the Pug should restock it, but I think it'd probably have to scale much faster than just adding two. (as in quadrupling the number each time) I think they'd also start implementing either auto-destruct or fight-until-dead (no disable) type technology.

**Comment by KiLEdEnNis** [comment]
how about we just create a mission branch that if you kill both Arfectas, the Unfettered invade more, talking to the Pug gets them to say "you fucked it up, you fix it", and you have to stop the Unfettered assault all alone
seems more realistic to me

**Comment by Zitchas** [comment]
lol. Yeah, that does. Or all the Wanderers get teleported out to somewhere else, a few systems are vanished to prevent the player from using the JD to escape in other directions, and the only route left is making it through a massive unfettered fleet that is completely swarming every remaining Wanderer and Unfettered system. And the only way to win and progress is if one has actually eliminated all of them.

**Comment by temtemy** [comment]
@KiLEdEnNis Sorry, I disagree. The Pug treats the Wanderers as their child, and we know that the Pug will do what is best for the Wanderers (like making them remember to fight). If I am a Pug, I won't allow a heartless monster (the player) to kill my beloved Wanderers. Sure, the player may be done with the Unfettered, who will they kill next? If they can kill innocent Pug, then surely the Wanderers are their next target. And I (again, if I am a Pug) won't allow it.

Four Arfectas are enough to make farming very challenging. The gridfire turrets are very strong and piercing, and with the improvements to the Arfecta AI and its combat effectiveness, they are more than enough to wipe out an Earth defense fleet (tier 1), can take out a tier 2 fleet as long as they cooperate by focusing fire one ship at a time, and probably a match against a tier 3 fleet, again by focusing fire and cloaking when enough damage is received. Not to mention that the Wanderers will attack you if you try to cap the Arfectas.

**Comment by temtemy** [comment]
@Zitchas And if the player has a cloaking device, then escaping Wanderer space is not a big problem.

**Comment by Amazinite** [comment]
>Four Arfectas are enough to make farming very challenging.

You underestimate the player. If one captures three Arfectas and causes four more to spawn, that could very easily just be another four Arfectas for the player's fleet, bringing them up to seven. Then what? Do we spawn more? The problem is that owning Arfectas creates a snowball effect where getting the next one becomes easier through being able to quickly disable it.

**Comment by temtemy** [comment]
> You underestimate the player. If one captures three Arfectas and causes four more to spawn, that could very easily just be another four Arfectas for the player's fleet, bringing them up to seven.

That makes sense, only if this PR didn't made the Arfecta very very difficult to capture (0.1% chance in a perfect setup Bactrian). That's why I made the suggestion here, not a separate issue. Even if the player managed to defeat the four Arfectas, most players will be discouraged to cap them, just like being discouraged to cap a World-Ship.

Or maybe instead of four, add eight? That would be almost impossible to defeat with just three Arfectas. The player would have to grind to be able to defeat those Arfectas, like making the Quarg permanently hostile to you and stealing their skylances. And probably most players won't want to anger the Quarg, because they might be locked out of a storyline.

Still, it doesn't make sense that the Pug will not react after you destroy the two Arfectas. Either take my suggestion, or KiLEdEnNis's.

**Comment by jafdy** [comment]
Because of the uncapped fleet sizes there is no upper limit to what the player can beat. Well, maybe their computer.
Another alternative is give the replacement arfectas better cloaks that will cloak them instantly, they could still be disabled but that would need to be done in a single frame from wherever they would normally cloak out.

**Comment by KiLEdEnNis** [comment]
my point wasn't meant to be taken by the example, my point was: you kill the Arfectas, so you get a small mission chain to clean up your mess, recoil from your actions
whatever you make the player do during those missions is left to whoever writes them

**Comment by Wrzlprnft** [comment]
thought: what if a fully stocked bactrian is no longer the top of the ladder? adjust every time?

**Comment by temtemy** [comment]
@KiLEdEnNis Then how can the player clean up their mess?

**Comment by Zitchas** [comment]
Do enough missions for the Pug to convince the Pug that the player is contrite, regrets the error of their ways, and wants to atone for it. 

Given the Pug's capabilities (they really don't need the player's help for anything), they may opt for something silly, debasing, or boring as a punishment. Ex. Kill a thousand drones that have been scattered throughout known space  (a handful in every system in human/hai/wanderer/korath space). Or make a delivery of some strange packages to each planet in human space. Neither example would be hard for the player to do, but it'd be long and frustrating. The point would be that it is insultingly simple, and obvious that the Pug are just creating something for them to do to prove themselves.

**Comment by Bladewood** [comment]
@Zitchas Story-wise that makes sense, but remember that Endless Sky is a game. [There's a reason why almost all of the old human licenses were removed.](https://github.com/endless-sky/endless-sky/commit/7a5bde4c4bb948772f918d9e70c600818138bd07)

**Comment by KiLEdEnNis** [comment]
> @KiLEdEnNis Then how can the player clean up their mess?

did you read
> whatever you make the player do during those missions is left to whoever writes them

??
I'm not planning to make such missions, so I'm not gonna tell someone else, if anyone else wants to, what I think it should be, assuming I even think of a general concept

**Comment by endless-sky** [comment]
This doesn't seem necessary to me - only a certain type of player will try to capture those Arfectas, and for that particular sort of player it will be more fun for them if the game lets them get away with it.

**Comment by temtemy** [comment]
@endless-sky nooooooo the drak > pug propaganda will intensify in r/endlesssky

**Comment by Turtleroku** [comment]
Is there a way to have 20 arfecta warp in (or uncloak) if you become hostile to the arfecta defending the wanderers?

**Comment by tehhowch** [comment]
@turtleroku all changes to game state for a player are done through missions and the associated `MissionAction`s / `GameEvent`s.

This PR is not the place to ask for help in achieving a desired outcome (ask in Google Groups, Steam, or Discord). Note that without #3005, no mission ships can spawn  until the player next lands or jumps.

**Comment by 10010101001** [comment]
Keep the 36 crew but increase it to 36 bio-defenses. Make the Arfectas in Wanderer space(only Arfectas in Wanderer space) have 1 attitude towards Wanderers and vice versa by adding a "_Pug_" faction.

File: data/wanderers start.txt [hunk 1]
```diff
@@ -2435,10 +2438,13 @@ mission "Wanderers Ap'arak 3: Pug"
 	to offer
 		has "Wanderers Ap'arak 2: done"
 	npc kill
-		personality staying uninterested
+		personality staying unconstrained uninterested
 		government "Pug"
 		ship "Pug Arfecta" "Hork Ekekek Ptui"
 		ship "Pug Arfecta" "Graak Sput Kchoo"
+	on complete
+		dialog `Shortly after defeating the Pug ships, you receive a message from Sobari Tele'ek: "We are [aware?, incredulous?] of reports that you have [exterminated, eradicated] our Keepers in the Ap'arak system. This is unacceptable! From now on, you are no longer welcome in our [territory, space]."`
```
Comment (Bladewood) [inline data/wanderers start.txt]: Question-marked terms in Wanderer translations should not be followed by commas.

**Comment by Amazinite** [comment]
The unlikely has become likely.

**Comment by Pointedstick** [comment]
Note that "unlikely" is not a synonym of "impossible." :)

**Pull request merged**